### PR TITLE
DAOS-7174 doc: add label support to admin and user guide

### PR DESCRIPTION
**Pull request by johannlombardi**
- Change command lines in both admin and user guide to
  use labels for both pools and containers instead of UUID.
- Add workflow section to user documentation
- Rename POSIX to filesystem and provide more details in the
  dfuse section.
- Remove references to `daos_server network scan` in favor of
  `dmg network scan`
- Several doc cleanups.

A few dmg and daos utility interface changes:
- support 'daos cont ls/list' to list containers
- support lsattr/delattr/setattr for attributes
- implement other useful shortcuts

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

File: doc/user/filesystem.md [hunk 1]
```diff
@@ -169,33 +168,91 @@ Additionally, there are several optional command-line options:
...
+
+#### Single Container Mode
+
+That's the most common use case where a pool and a POSIX container are provided
+on the command line. The moun point will map to the root of the container
+itself. Files can be accessed by simply concatenating the mount point and the
+name of the file, relative to the root of the container.
+
+```bash
+$ daos cont create tank -l mycont -t POSIX
```
[the remark this answers is not in the document]
Comment (mchaarawi) [reply doc/user/filesystem.md:190]: nevermind, i realized that this is the pool

File: doc/user/filesystem.md [hunk 2]
```diff
@@ -207,10 +264,10 @@ To create a new container and link it into the namespace of an existing one,
 use the following command.
 
 ```bash
-$ daos container create --type POSIX --pool <pool uuid> --path <path to entry point>
+$ daos container create --type POSIX <pool_label> --path <path to entry point>
```
Comment (mchaarawi) [inline doc/user/filesystem.md:267]: probably better to move the label after the create, before --type posix

File: doc/user/container.md [hunk 3]
```diff
@@ -1,111 +1,245 @@
...
+
+The table below summarizes the different container properties available.
+
+| **Container Property**  | **Immutable**   | **Description** |
+| ------------------------| --------------- | ----------------|
+| label			  | No              | String associate with a containers. e.g., "Cat\_Pics" or "ResNet50\_training\_data".|
+| owner                   | Yes             | User acting as the owner of the container.|
+| group                   | Yes             | Group acting as the owner of the container|
+| layout\_type            | Yes             | The container type (POSIX, HDF5, ...)|
+| layout\_ver`            | Yes             | A version of the layout that can be used by I/O middleware to handle interoperability.|
```
Comment (mchaarawi) [inline doc/user/container.md:215]: extra `?

File: doc/user/container.md [hunk 4]
```diff
@@ -1,111 +1,245 @@
...
+| acl                     | No              | Container access control list.|
+| compression             | Yes             | Whether online compression is enabled (off, lz4, deflate[1-4])|
+| dedup                   | Yes             | Inline deduplication off, hash based (hash) or using memory compare (memcmp)|
+| dedup_threshold         | Yes             | Minimum I/O size to consider for deduplication|
+| encryption              | Yes             | Inline encryption off, or algorithm
+to use (XTS[128|256], CBC[128|192|256] or GCM]129|256]|
+| status                  | No              | Current state of the container|
+| alloc_oid               | No              | Maximum allocated object ID by container allocator|
+
+Refer to the Data Integrity and Access Control Lists sections for more
```
Comment (mchaarawi) [inline doc/user/container.md:232]: always useful to have links there

File: doc/user/workflow.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
+# Workflow
+
+# Use Cases
+
+A DAOS pool is a persistent storage reservation that is allocated to a
+project or specific job. Pools are allocated, shrunk, grew and destroyed by
```
Comment (mchaarawi) [inline doc/user/workflow.md:6]: grew -> grown

File: doc/user/workflow.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+the administrators.
+
+The typical workflow consists of:
+* New project members meet and define storage requirements including space,
+bandwith, IOPS & data protection needs.
+* Administrators collect those requirements, create a pool for the new
+project and set relevant ACL to grant access to project members.
+* Administrators notify the project members that the pool has been created and
```
Comment (mchaarawi) [inline doc/user/workflow.md:14]: notify->notifies

File: doc/user/workflow.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+provide the pool label to the users.
```
Comment (mchaarawi) [inline doc/user/workflow.md:15]: provide->provides

**Review by mchaarawi — APPROVED** [review]

File: doc/user/filesystem.md [hunk 8]
```diff
@@ -169,33 +168,91 @@ Additionally, there are several optional command-line options:
...
-launched. How POSIX containers are represented within that mount point varies 
-depending on the DFuse command-line options:
+launched. How POSIX containers are represented within that mount point varies
+depending on the DFuse command-line options.
+DFuse can operate in three modes.
+
+#### Single Container Mode
+
+That's the most common use case where a pool and a POSIX container are provided
+on the command line. The moun point will map to the root of the container
```
[the remark this answers is not in the document]
Comment (johannlombardi) [reply doc/user/filesystem.md:185]: fixed.

File: doc/user/container.md (hunk 3 above)
Comment (johannlombardi) [reply doc/user/container.md:215]: good catch.

File: doc/user/filesystem.md (hunk 2 above)
Comment (johannlombardi) [reply doc/user/filesystem.md:267]: fixed.

File: doc/user/workflow.md (hunk 5 above)
Comment (johannlombardi) [reply doc/user/workflow.md:6]: fixed

File: doc/user/workflow.md (hunk 6 above)
Comment (johannlombardi) [reply doc/user/workflow.md:14]: i think this one is correct ...

File: doc/user/workflow.md (hunk 7 above)
Comment (johannlombardi) [reply doc/user/workflow.md:15]: same here?

File: doc/user/filesystem.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (johannlombardi) [reply doc/user/filesystem.md:190]: https://serverfault.com/questions/562564/why-are-all-the-zpools-named-tank

File: doc/admin/pool_operations.md [hunk 9]
```diff
@@ -3,50 +3,55 @@
...
+
 [create command options]
       -g, --group=     DAOS pool to be owned by given group, format name@domain
       -u, --user=      DAOS pool to be owned by given user, format name@domain
-      -p, --name=      Unique name for pool (set as label)
+      -p, --label=     Unique label for pool
+      -P, --properties= Pool properties to be set
       -a, --acl-file=  Access Control List file path for DAOS pool
       -z, --size=      Total size of DAOS pool (auto)
       -t, --scm-ratio= Percentage of SCM:NVMe for pool storage (auto) (default: 6)
```
Comment (tanabarr) [inline doc/admin/pool_operations.md:43]: needs updating after tiering lands

File: doc/admin/pool_operations.md [hunk 10]
```diff
@@ -64,92 +69,125 @@ with redundancy enabled by default (pool service replicas on ranks 1-3).
...
-`
+$ dmg pool evict tank
+Pool-evict command succeeded
+```
 
 **To see a list of the pools in your DAOS system:**
 
 ```bash
-$ dmg system list-pools
+$ dmg pool list
```
Comment (tanabarr) [inline doc/admin/pool_operations.md:82]: needs updating now ffccb10d9bb6f49c9780d3442575334891e156e0 has landed

File: doc/user/filesystem.md [hunk 11]
```diff
@@ -114,33 +118,28 @@ vice versa.
...
+except symbolic links, are not supported, nor are any ACLs.
 
 DFuse can run in the foreground, keeping the terminal window open, or it can
-daemonize to run like a system daemon. 
-However, to do this and still be
-able to access DAOS it needs to daemonize before calling `daos_init()`. 
-This in turns means it cannot report some kinds of startup errors either on
-stdout/stderr or via its return code.  
-When initially starting with DFuse it is recommended to run in foreground mode 
+daemonize to run like a system daemon.
```
Comment (tanabarr) [inline doc/user/filesystem.md:131]: suggest replacing "like a system daemon" with "in the background"

File: doc/user/filesystem.md [hunk 12]
```diff
@@ -169,33 +168,91 @@ Additionally, there are several optional command-line options:
 
 When DFuse starts, it will register a single mount with the kernel, at the
 location specified by the `--mountpoint` option. This mount will be
-visible in `/proc/mounts`, and possibly in the output of `df`.  
-The contents of multiple pools/containers will be accessible via this 
+visible in `/proc/mounts`, and possibly in the output of `df`.
```
Comment (tanabarr) [inline doc/user/filesystem.md:171]: maybe we could elaborate on what "possibly" means in this context, when will we not see the mount when calling df?

File: doc/user/workflow.md (hunk 6 above)
Comment (mchaarawi) [reply doc/user/workflow.md:14]: yea sorry, i read as administrator

**Review by mchaarawi — APPROVED** [review]

**Comment by cdurf1** [comment]
Administration.md file will not render properly as-is. The other docs are OK. I'll edit the admin file and add a commit.

**Review by tanabarr — COMMENTED** [review]
None of the comments are blocking, just suggestions. the only actual inaccuracy is the `dmg pool list` command output.

**Review by tanabarr — APPROVED** [review]

**Review by cdurf1 — CHANGES_REQUESTED** [review]
Approved with changes to administration.md file. Tables now render appropriately. edited file committed to your branch

File: doc/user/filesystem.md [hunk 13]
```diff
@@ -250,10 +331,10 @@ The following types of data will be cached by default.
 * Readahead in dfuse and inserting data into kernel cache
 * MMAP write optimization
 
-!note
-Caching is enabled by default in dfuse. This might cause some parallel
-applications to fail. Please disable caching if you experience this or want
-up to date data sharing between nodes.
+!!! note
+    Caching is enabled by default in dfuse. This might cause some parallel
+    applications to fail. Please disable caching if you experience this or want
```
[the remark this answers is not in the document]
Comment (johannlombardi) [reply doc/user/filesystem.md:336]: fixing that in another PR.

File: src/control/cmd/daos/container.go [hunk 14]
```diff
@@ -32,32 +32,32 @@ import "C"
 
 type containerCmd struct {
 	Create      containerCreateCmd      `command:"create" description:"create a container"`
+	List        containerListCmd        `command:"list" alias:"ls" description:"list all containers in pool"`
```
[the remark this answers is not in the document]
Comment (johannlombardi) [reply src/control/cmd/daos/container.go:35]: It is mentioned in the PR commit message:

"A few dmg and daos utility interface changes:

- support 'daos cont ls/list' to list containers
- support lsattr/delattr/setattr for attributes
- implement other useful shortcuts"

**Pull request merged**